hs/app/reachpc/ReachPC/Config.hs

```diff
@@ -1,27 +1,29 @@
-{- HLINT ignore "Use newtype instead of data" -}
-{-# LANGUAGE TemplateHaskell #-}
+{- HLINT ignore "Use if" -}
+{-# LANGUAGE AllowAmbiguousTypes #-}
```

Comment:
Why do you need this? It has a scary name. @DanBurton ?

Reply:
It allows `showEnum`. The compiler suggested it. I don't 100% understand it but it doesn't seem bad.

Reply:
https://stackoverflow.com/questions/45128492/what-is-allowambiguoustypes-and-why-is-it-needed-in-this-forall-example somewhat explains it for me

Reply:
`showEnum` feels very powerful and a general tool separate from `ReachPC.Config`'s purview

Reply:
I don't think it's that crazy. It just makes a list of all the variants of the type you give it. Handily, when deriving all the needed instances, it gives you `Read`-able strings that can be substituted for user input. It just saves a few lines. I was inspired by Tomland.enumBounded. I don't think `AllowAmbiguousTypes` is that crazy either, my understanding is that ghc doesn't like inferring a type from an already inferred type. The flag makes that type inference lazier. It doesn't change program safety.

Reply:
If you want I can just make it take a Proxy or make them literals.

Reply:
I want @DanBurton 's opinion on adding the extension (at the whole project level or just here) or doing the proxy

Reply:
I don't like it but I don't hate it. Seems harmless but feels icky. But having to write Proxy is ickier. Let's keep it enabled only at the module level rather than whole project. Maybe I will become desensitized to it over time and be ok with it being enabled for the whole project.

Reply:
Ok